src/utils/proofs_api/paramfetch.rs

```diff
@@ -171,53 +169,47 @@ async fn fetch_verify_params(
 
     fetch_params(&path, &info).await?;
 
-    check_file(path, info).await.map_err(|e| {
+    check_file(&path, &info).await.map_err(|e| {
         // TODO remove invalid file
         e.into()
     })
 }
 
-async fn fetch_params(path: &Path, info: &ParameterData) -> Result<(), anyhow::Error> {
+async fn fetch_params(path: &Path, info: &ParameterData) -> anyhow::Result<()> {
+    let cid = Cid::from_str(&info.cid)?;
     let gw = std::env::var(GATEWAY_ENV).unwrap_or_else(|_| GATEWAY.to_owned());
     info!("Fetching param file {:?} from {}", path, gw);
-    let url = format!("{}{}", gw, info.cid);
     let result = retry(ExponentialBackoff::default(), || async {
-        Ok(fetch_params_inner(&url, path).await?)
+        Ok(async {
+            let tmp =
+                tempfile::NamedTempFile::new_in(path.parent().unwrap_or_else(|| Path::new(".")))?;
+            download_ipfs_file_trustlessly(&cid, Some(GATEWAY), tmp.path()).await?;
+            // The extra check is redundant, just making sure `rs-car-ipfs` is bug free, consider removing it in the future.
+            check_file(tmp.path(), info).await?;
```

Comment:
It makes me doubt the crate, honestly. That is to say, if we don't have enough trust in the crate that we write runtime tests for it, then maybe we don't want it really given we have an alternative.

Reply:
Did more investigation, according to the [doc](https://docs.rs/futures/latest/futures/io/struct.BufWriter.html) the `BufWriter` needs to be manually flushed before being dropped. So the issues here and in the previous reverted PR are misuses of `BufWriter` and have nothing to do with the `rs-car-ipfs` crate.

Reply:
Regarding the check, I'm inclined to keep it for now as we already have the digests of these files